common/Windows/SecondaryWindow.cs

```diff
@@ -0,0 +1,328 @@
+                {
+                    OnThemeChanged(null, ThemeSelector.Theme);
+                    ThemeSelector.ThemeChanged += OnThemeChanged;
+                }
+                else
+                {
+                    ThemeSelector.ThemeChanged -= OnThemeChanged;
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the window should be modal.
+    /// </summary>
+    /// <remarks>Setting this property to true will disable interaction on the primary window.</remarks>
+    public bool IsModal
+    {
+        get => _isModal;
+        set
+        {
+            if (_isModal != value)
+            {
+                _isModal = value;
+                if (PrimaryWindow != null)
+                {
+                    // Note: This is a temporary workaround until there's a
+                    // built-in support for modal windows.
+                    PInvoke.EnableWindow((HWND)PrimaryWindow.GetWindowHandle(), !value);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the secondary window should stay on top of the primary window.
+    /// </summary>
+    public bool IsTopLevel
+    {
+        get => _isTopLevel;
+        set
+        {
+            if (_isTopLevel != value)
+            {
+                _isTopLevel = value;
+
+                // Note: Setting the owner here is a temporary workaround until there's is
+                // a built-in support for creating secondary windows with an owner while
+                // also being able to customize the content from XAML.
+                // Related: https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/windowing/windowing-overview#limitations
+                // "The Windows App SDK doesn't currently provide methods for attaching UI framework content to an AppWindow."
+
+                // Set primary window as owner of secondary window
+                var sWindow = (HWND)this.GetWindowHandle();
+                var pWindow = (HWND?)PrimaryWindow?.GetWindowHandle() ?? HWND.Null;
+                SetWindowOwner(sWindow, value ? pWindow : HWND.Null);
+            }
+        }
+    }
+
+    public WindowEx? PrimaryWindow
+    {
+        get => _primaryWindow;
+        set
+        {
+            if (_primaryWindow != value)
+            {
+                // If a previous primary window was connected, disconnect it
+                if (_primaryWindow != null)
+                {
+                    _primaryWindow.Closed -= OnPrimaryWindowClosed;
+                }
+
+                // Set new primary window
+                _primaryWindow = value;
+
+                // If a primary window is set, connect it
+                if (_primaryWindow != null)
+                {
+                    _primaryWindow.Closed += OnPrimaryWindowClosed;
+                }
+            }
+        }
+    }
+
+    public SecondaryWindow()
```

Comment:
Good question! The answer is yes technically we can. However, I tend to not register UI component (except maybe Views as mentioned in the case of MAUI [pdf book page 22](https://learn.microsoft.com/en-us/dotnet/architecture/maui/dependency-injection)), although we could also consider the window a view as  well (maybe?). For now, I will keep `SecondaryWindow` unregistered so that the derived classes (`InstallationNotesWindow` and `DevDriveWindow`) don't have to worry about passing those dependencies to the base class (which those would also be registered in the container to enable explicitly passing the injected parameters as arguments to the base constructor). Let me know if you have suggestion or concerns.

Reply:
I was thinking of it as a view. but I see how there is more work that would be needed to make that work. No issues thanks :)